python/pyspark/sql/session.py

```diff
@@ -301,6 +306,9 @@ def __init__(self, sparkContext: SparkContext, jsparkSession: Optional[JavaObjec
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
             else:
                 jsparkSession = self._jvm.SparkSession(self._jsc.sc())
```

Comment:
LGTM with a couple of nits: @AngersZhuuuu,

- Can we actually leverage existing constructor on `SparkSession` to pass the initial options instead of manually? Here unlike Scala, it initiates `sharedState` always. I think it's best to keep the code path matched.
- Another nit is that: It's always preferred to use less Py4J connections which exposes potential flakiness.

Reply:
> Can we actually leverage existing constructor on SparkSession to pass the initial options instead of setting it manually? Here unlike Scala, it initiates sharedState always. I think it's best to keep the code path matched.

Yea, will try this.

Reply:
Thanks!

Reply:
> 

Create a new one or a followup?

Reply:
Yup, let's create a followup PR.